test/e2e/framework/providers/factory.go

```diff
@@ -88,8 +88,10 @@ func SetupProviderConfig(providerName string) (ProviderInterface, error) {
 // ProviderInterface contains the implementation for certain
 // provider-specific functionality.
 type ProviderInterface interface {
-	FrameworkBeforeEach(f *Framework)
-	FrameworkAfterEach(f *Framework)
+	// TODO(WanLinghao): Revert this function, use Framework directly
+	// see: https://github.com/kubernetes/kubernetes/pull/81464
```

Comment:
Not sure I understand what's going on here?

Reply:
The thing is, factory.go is using pakcage `k8s.io/kubernetes/test/e2e/framework/`. And package framework is using factory.go. What  I am trying to do here is to break import cycle. I have no better way to do this..